ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/network/NodePermissioningAdapter.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.p2p.network;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import tech.pegasys.pantheon.ethereum.chain.Blockchain;
+import tech.pegasys.pantheon.ethereum.p2p.peers.Peer;
+import tech.pegasys.pantheon.ethereum.p2p.permissions.PeerPermissions;
+import tech.pegasys.pantheon.ethereum.permissioning.node.NodePermissioningController;
+import tech.pegasys.pantheon.ethereum.permissioning.node.provider.SyncStatusNodePermissioningProvider;
+import tech.pegasys.pantheon.util.enode.EnodeURL;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+class NodePermissioningAdapter extends PeerPermissions {
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final NodePermissioningController nodePermissioningController;
+  private final List<EnodeURL> bootnodes;
+  private final Blockchain blockchain;
+  private final long blockchainListenId;
+  private final long nodePermissioningListenId;
+
+  public NodePermissioningAdapter(
+      final NodePermissioningController nodePermissioningController,
+      final List<EnodeURL> bootnodes,
+      final Blockchain blockchain) {
+    checkNotNull(nodePermissioningController);
+    checkNotNull(bootnodes);
+    checkNotNull(blockchain);
+
+    this.nodePermissioningController = nodePermissioningController;
+    this.bootnodes = bootnodes;
+    this.blockchain = blockchain;
+
+    // TODO: These events should be more targeted
+    blockchainListenId =
+        blockchain.observeBlockAdded((evt, chain) -> dispatchUpdate(true, Optional.empty()));
+    nodePermissioningListenId =
+        this.nodePermissioningController.subscribeToUpdates(
+            () -> dispatchUpdate(true, Optional.empty()));
```

Comment:
This event never gets fired in tests.  Is this something that the Dapp will likely exercise?

Reply:
The permissioninng dapp doesn't play a role in this flow. Our `NodePermissioningController` needs to subscribe to chain updates so we can verify if a new block must trigger a re-evaluation of the rules (e.g. the block contains a transaction that removes a node from the whitelist).

Reply:
We have acceptance tests that will hit this branch.  The `NodePermissioningController` is in a separate package, so I don't want to pull in a real instance and try figuring out how to get that to broadcast an update within the p2p test suite.  Testing for that object should be covered in the `permissioning` package.